src/context/serverStateSlice.ts

```diff
@@ -0,0 +1,34 @@
+import { Map } from 'immutable';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { RootState } from './store';
+import { AvailableDates } from '../config/types';
+import { LAYERS_AVAILABLE_DATES } from '../constants';
+
+interface ServerState extends Map<string, any> {}
+
+const initialState: ServerState = Map({
+  availableDates: Map(
+    JSON.parse(localStorage.getItem(LAYERS_AVAILABLE_DATES) || '[]'),
+  ) as AvailableDates,
+});
```

Comment:
Is this an expensive call? It seems like it just adds a bunch of unnecessary state & complexity to bother with local storage, why not just init this on load?